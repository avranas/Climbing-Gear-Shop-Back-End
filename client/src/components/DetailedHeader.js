import { Link } from 'react-router-dom';
import SearchBar from './SearchBar/SearchBar';
import HeaderUserButtons from './HeaderUserButtons';
import { useDispatch, useSelector } from 'react-redux';
import { selectUser } from "../slices/userSlice";
import { useEffect } from 'react';
import axios from '../api/api';

const DetailedHeader = (props) => {

  const user = useSelector(selectUser);
  const dispatch = useDispatch();

  useEffect(() => {
    const getUserData = async () => {
      try {
        const response = await axios.get('/user');
        dispatch({type: 'user/loadUserData', payload: response.data});
      } catch (err) {
        console.log(err);
      }
    }
    if (user.loggedIn) {
      getUserData();
    }
  });

  return (
    <header className='container'>
      <Link to="/">
        <div className="logo" />
      </Link>
      <SearchBar />
      {
        user.loggedIn &&
        <p>{`Welcome, ${user.firstName}`}</p>
      }
      <HeaderUserButtons />
      <Link to='/cart'>
        <div id="cart-icon" />
      </Link>
    </header>
  );
};

export default DetailedHeader;
