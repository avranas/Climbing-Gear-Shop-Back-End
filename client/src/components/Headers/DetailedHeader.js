import { Link } from "react-router-dom";
import SearchBar from "../SearchBar/SearchBar";
import HeaderUserButtons from "../HeaderButtons/HeaderButtons";
import { useDispatch, useSelector } from "react-redux";
import { selectUser } from "../../slices/userSlice";
import { useEffect, useState } from "react";
import "./Header.css";
import { selectCart, loadCartData } from "../../slices/cartSlice";
import axios from "axios";

const DetailedHeader = (props) => {
  const cart = useSelector(selectCart);
  const cartData = cart.data;
  const dispatch = useDispatch();
  const [loggedIn, setLoggedIn] = useState(false);
  const user = useSelector(selectUser);

  useEffect(() => {
    const getUserData = async () => {
      try {
        const response = await axios("/authenticated");
        if (response.data) {
          const response = await axios.get("/user");
          setLoggedIn(true);
          dispatch({ type: "user/loadUserData", payload: response.data });
        }
      } catch (err) {
        console.log(err);
      }
    };
    loadCartData(dispatch);
    getUserData();
  }, [dispatch]);
  let firstName = "";
  if (user.name) {
    firstName = user.name.split(" ")[0];
  }

  return (
    <header className="header">
      <div id="detailed-header-content">
        <Link to="/">
          <div id="detailed-header-logo" className="logo header-item" />
        </Link>
        <div className="header-item">
          <SearchBar />
        </div>
        {loggedIn && user.name ? (
          <div className="header-item" id="welcome-message">
            <p>Welcome,</p>
            <p>{firstName}</p>
          </div>
        ) : (
          <div className="header-item" id="welcome-message">
            {" "}
          </div>
        )}
        <div id="header-user-buttons-container" className="header-item">
          <HeaderUserButtons />
        </div>
        <div className="header-item">
          <Link to="/cart">
            <div id="cart-icon">
              {cartData.itemCount > 99 ? (
                <div className="cart-items-count" id="double-digit">
                  99+
                </div>
              ) : cartData.itemCount > 9 ? (
                <div className="cart-items-count" id="double-digit">
                  {cartData.itemCount}
                </div>
              ) : cartData.itemCount > 0 ? (
                <div className="cart-items-count" id="single-digit">
                  {cartData.itemCount}
                </div>
              ) : null}
            </div>
          </Link>
        </div>
      </div>
    </header>
  );
};

export default DetailedHeader;
