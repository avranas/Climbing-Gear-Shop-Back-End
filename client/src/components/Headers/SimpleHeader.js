import { Link } from 'react-router-dom';

const SimpleHeader = (props) => {

  return (
    <header className='header' id='simple-header-content'>
        <Link to="/">
          <div className="logo" />
        </Link>
    </header>
  );
};

export default SimpleHeader;
