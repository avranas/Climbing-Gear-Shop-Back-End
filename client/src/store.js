import { configureStore } from "@reduxjs/toolkit";
import userReducer from './slices/userSlice';
import productListSlice from './slices/productListSlice';
import notificationSlice from './slices/notificationSlice';
import productSlice from "./slices/productSlice";
import cartSlice from "./slices/cartSlice";
import newestCartItemSlice from "./slices/newestCartItemSlice";

export default configureStore({
  reducer: {
    user: userReducer,
    productList: productListSlice,
    product: productSlice,
    notifications: notificationSlice,
    cart: cartSlice,
    newestCartItem: newestCartItemSlice
  },
});