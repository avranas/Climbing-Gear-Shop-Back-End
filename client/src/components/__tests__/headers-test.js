import { cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import store from "../../store";
import SimpleHeader from "../Headers/SimpleHeader";
import DetailedHeader from "../Headers/DetailedHeader";

afterEach(() => {
  cleanup();
});

const simpleHeader = (
  <Provider store={store}>
    <BrowserRouter>
      <SimpleHeader />
    </BrowserRouter>
  </Provider>
);

const detailedHeader = (
  <Provider store={store}>
    <BrowserRouter>
      <DetailedHeader />
    </BrowserRouter>
  </Provider>
);

test("SimpleHeader matches snapshot", async () => {
  const tree = renderer.create(simpleHeader).toJSON();
  expect(tree).toMatchSnapshot();
});

test("DetailedHeader matches snapshot", async () => {
  const tree = renderer.create(detailedHeader).toJSON();
  expect(tree).toMatchSnapshot();
});