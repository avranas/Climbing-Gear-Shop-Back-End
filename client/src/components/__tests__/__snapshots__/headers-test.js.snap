// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedHeader matches snapshot 1`] = `
<header
  className="header"
>
  <div
    id="detailed-header-content"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div
        className="logo header-item"
        id="detailed-header-logo"
      />
    </a>
    <div
      className="header-item"
    >
      <div
        data-testid="searchbar"
        id="search-bar"
      >
        <input
          id="search-text"
          name="search"
          onChange={[Function]}
          onKeyUp={[Function]}
          type="text"
          value=""
        />
        <button
          data-testid="searchbar-button"
          id="search-button"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="header-item"
      id="welcome-message"
    >
       
    </div>
    <div
      className="header-item"
      id="header-user-buttons-container"
    >
      <div
        id="header-user-buttons"
      >
        <div />
      </div>
    </div>
    <div
      className="header-item"
    >
      <a
        href="/cart"
        onClick={[Function]}
      >
        <div
          id="cart-icon"
        />
      </a>
    </div>
  </div>
</header>
`;

exports[`SimpleHeader matches snapshot 1`] = `
<header
  className="header"
  id="simple-header-content"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <div
      className="logo"
    />
  </a>
</header>
`;
